paint/data/stac_client.py

```diff
@@ -0,0 +1,560 @@
+        """
+        self.output_dir = pathlib.Path(output_dir)
+        if not self.output_dir.is_dir():
+            self.output_dir.mkdir(parents=True, exist_ok=True)
+        self.chunk_size = chunk_size
+
+    @staticmethod
+    def get_catalog(href: str) -> pystac.catalog.Catalog:
+        """
+        Help to load a STAC catalog.
+
+        Since pystac does not check if the URL exists before attempting to download a catalog we catch any errors raised
+        and raise a value error indicating that the desired catalog does not exist.
+
+        Parameters
+        ----------
+        href : str
+            URL reference to the catalog that should be accessed.
+
+        Returns
+        -------
+        pystac.catalog.Catalog
+            Catalog, if it exists.
+
+        Raises
+        ------
+        ValueError
+            If the URL does not exist.
+        """
+        try:
+            return pystac.Catalog.from_file(href=href)
+        except Exception as e:
+            # Log the error message for inspection.
+            error_message = str(e)
+            # Handle the 404 error explicitly by checking the message content.
+            if "404" in error_message or "Could not read uri" in error_message:
+                raise ValueError(
+                    f"The desired catalog with the URL '{href}' doesn't exist (404 error)!"
+                )
+            else:
+                # Raise the original exception for all other errors
+                raise
+
+    @staticmethod
+    def get_child(
+        parent: Union[pystac.catalog.Catalog, pystac.collection.Collection],
+        child_id: str,
+    ) -> Union[pystac.catalog.Catalog, pystac.collection.Collection, pystac.item.Item]:
+        """
+        Help to get a child from a STAC catalog or collection.
+
+        Since pystac returns `None` if a child is not present, it is important we log a warning if this is the case.
+
+        Parameters
+        ----------
+        parent : Union[pystac.Catalog, pystac.Collection]
+            Parent STAC catalog or collection.
+        child_id : str
+            ID of the child STAC catalog, collection, or item.
+
+        Returns
+        -------
+        Union[pystac.Catalog, pystac.Collection, pystac.Item]
+            Child STAC catalog, collection, or item.
+        """
+        child = parent.get_child(child_id)
+        if child is None:
+            log.warning(
+                f"The child with ID {child_id} is not available, data for this child will not be downloaded."
+            )
+        return child
+
+    def download_file(self, url: str, file_name: Union[str, pathlib.Path]) -> None:
+        """
+        Download a file.
+
+        Parameters
+        ----------
+        url : str
+            URL of the file to download.
+        file_name : Union[str, pathlib.Path]
+            File name to be saved.
+        """
+        response = requests.get(url, stream=True)
```

Comment:
```suggestion
        # Send an HTTP GET request to the specified url to retrieve the file.
        # The `stream=True` parameter tells requests to download the file in chunks instead of loading the entire file into memory.
        response = requests.get(url, stream=True)
```